### PR TITLE
Add azurerm_log_analytics_workspace resource

### PR DESCRIPTION
**Pull request by sergenyalcin**
Signed-off-by: sergenyalcin <yalcinsergen97@gmail.com>

### Description of your changes

Implementing azurerm_log_analytics_workspace resource

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

- Adding a workspace via MR
- Updating the existing workspace
- Deleting the workspace

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by ulucinar** [comment]
Hi @sergenyalcin,
Thank you for the PR! Let's merge the base PR #83 and then rebase this PR.

**Comment by ulucinar** [comment]
Hi @sergenyalcin,
This PR is ready to be rebased in the way we discussed offline. 
Cheers :)

**Comment by ulucinar** [comment]
Hi @sergenyalcin again :)
Could you please rebase onto #88 (the renaming PR)?

File: config/log/config.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+const groupLog = "loganalytics"
+
+// Configure configures loganalytics group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_log_analytics_workspace", func(r *config.Resource) {
+		r.Kind = "Workspace"
+		r.ShortGroup = groupLog
```
Comment (ulucinar) [inline config/log/config.go:32]: Do all related resources for the "OperationalInsights" service have Terraform resource names starting with "azurerm_log_analytics"? Could you please make a quick check?

File: docs/adding-resources.md [hunk 2]
```diff
@@ -34,10 +34,23 @@ configuration for the resource.
        make generate
        ```
 
-   6. Run against a Kubernetes cluster:
+   6. Register CRDs to cluster:
 
        ```console
        kubectl apply -f package/crds
+       ```
+   7. Run the provider against a Kubernetes cluster
```
Comment (ulucinar) [inline docs/adding-resources.md:42]: Thank you @sergenyalcin for figuring this out and correcting the documentation! Very much appreciated.

File: docs/adding-resources.md [hunk 3]
```diff
@@ -34,10 +34,23 @@ configuration for the resource.
...
+
+       ```console
+       make run TERRAFORM_VERSION=<terraform_version> TERRAFORM_PROVIDER_VERSION=<provider_version> TERRAFORM_PROVIDER_SOURCE=<provider_source>
```
Comment (ulucinar) [inline docs/adding-resources.md:45]: Specific to this repo:
```suggestion
       make run TERRAFORM_VERSION=<terraform_cli_version> TERRAFORM_PROVIDER_VERSION=<provider_version> TERRAFORM_PROVIDER_SOURCE=hashicorp/azurerm
```

File: docs/adding-resources.md [hunk 4]
```diff
@@ -34,10 +34,23 @@ configuration for the resource.
...
+       ```
+
+       or
+
+       ```console
+       export TERRAFORM_VERSION=<terraform_version>
+       export TERRAFORM_PROVIDER_VERSION=<provider_version>
+       export TERRAFORM_PROVIDER_SOURCE=<provider_source>
```
Comment (ulucinar) [inline docs/adding-resources.md:53]:
```suggestion
       export TERRAFORM_PROVIDER_SOURCE=<hashicorp/azurerm>
```

File: docs/adding-resources.md [hunk 5]
```diff
@@ -34,10 +34,23 @@ configuration for the resource.
...
+   7. Run the provider against a Kubernetes cluster
+
+       ```console
+       make run TERRAFORM_VERSION=<terraform_version> TERRAFORM_PROVIDER_VERSION=<provider_version> TERRAFORM_PROVIDER_SOURCE=<provider_source>
+       ```
+
+       or
+
+       ```console
+       export TERRAFORM_VERSION=<terraform_version>
```
Comment (ulucinar) [inline docs/adding-resources.md:51]:
```suggestion
       export TERRAFORM_VERSION=<terraform_cli_version>
```

File: examples/loganalytics/workspace.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: loganalytics.azure.jet.crossplane.io/v1alpha1
+kind: Workspace
+metadata:
+  name: example
```
Comment (ulucinar) [inline examples/loganalytics/workspace.yaml:4]:
```suggestion
  name: example-ws
```

File: config/log/config.go (hunk 1 above)
Comment (sergenyalcin) [reply config/log/config.go:32]: Hi @ulucinar I checked the related resources. All "OperationalInsights" service resources have Terraform resource names starting with "azurerm_log_analytics"

File: docs/adding-resources.md (hunk 2 above)
Comment (sergenyalcin) [reply docs/adding-resources.md:42]: Sure thing @ulucinar

**Review by ulucinar — APPROVED** [review]
Thank you @sergenyalcin!

**Pull request merged**